### PR TITLE
Native Disk Update

### PR DESCRIPTION
**Pull request by s4heid**
* marks the Native Disk Resize feature as supported on Azure (which was seemingly forgotten).
* marks the Native Disk Update feature as supported for Azure in the next CPI release (`v50.0.0`?)
* adds documentation which explains the newly introduced `update_disk` method
* moves the `disk_size` property outside of the `cloud_properties` block, as `disk_size` is not a cloud property.

### CODE REVIEW

**Review by MSSedusch — APPROVED** [review]
lgtm

**Comment by jpalermo** [comment]
Should wait for https://github.com/cloudfoundry/bosh/pull/2555 before merging

**Review by aramprice — APPROVED** [review]

**Review by ystros — APPROVED** [review]

**Comment by beyhan** [comment]
@rkoster, @aramprice I'm wondering what the status with the implementation of this is. It was reverted by https://github.com/cloudfoundry/bosh/pull/2570 because it failed the integration tests. Was there a revert of the revert?

**Comment by aramprice** [comment]
@beyhan - the rollback of https://github.com/cloudfoundry/bosh/pull/2555 has not been undone.

As noted in https://github.com/cloudfoundry/bosh/pull/2570 we rolled back because the integration test that was added did not run (and in fact could not have run successfully given the way the test was written). To my knowledge our team hasn't done any work to fix the integration spec or re-merge https://github.com/cloudfoundry/bosh/pull/2555.

Unfortunately there isn't currently a documented way for folks to run the integration specs locally which makes it hard for contributors to write these tests. For folks who have write access to https://bosh.ci.cloudfoundry.org/ there is `rake fly:integration` which will execute the integration specs via Concourse, so depending on who is making the contribution this may be an option. Check out `src/bosh-dev/lib/bosh/dev/tasks/fly.rake` in the BOSH repo for more context.

**Comment by beyhan** [comment]
@a-hassanin could someone of you who is an approval and has write access to https://bosh.ci.cloudfoundry.org/ support @s4heid  for running and debugging the integration test as suggested by @aramprice.

Another question I wanted to raise here is whether we should wait with the documentation until we have successfully integrated this feature. I think If we don't get the feature in the next week in we should revert the docs until we have it. @rkoster, @jpalermo any opinion on this?

**Pull request merged**
